crates/nu-parser/src/parse_keywords.rs

```diff
@@ -350,6 +352,15 @@ pub fn parse_for(working_set: &mut StateWorkingSet, spans: &[Span]) -> Expressio
     }
 }
 
+/// If `name` is a keyword, emit an error.
+pub fn verify_not_keyword(working_set: &mut StateWorkingSet, name: &str, span: Span) {
+    if RESERVED_VARIABLE_NAMES.contains(&name) {
+        // to_string is costly, but there would be clone here anyways, since
+        // name is taken inside the if clause.
```

Comment:
Good call to add a docstring about the side effect that an error will be created. Still as things stand the helper can be private to the module.

Maybe I am a bit pedantic about the naming:
```suggestion
/// If `name` is a reserved variable name, emit an error.
fn verify_not_reserved_variable_name(working_set: &mut StateWorkingSet, name: &str, span: Span) {
    if RESERVED_VARIABLE_NAMES.contains(&name) {
```

Thanks for being careful about allocations but that seems absolutely neglible in the grand scheme of things as it is only exercised in a very rare case for a small string.

Reply:
`pub` is an error, forgot to remove it, did that with the last push.